pkg/apis/api/types.go

```diff
@@ -800,10 +800,19 @@ type PullRequestOptions struct {
 }
 
 type ComponentReportRequestReleaseOptions struct {
-	Release            string
-	PullRequestOptions *PullRequestOptions
-	Start              time.Time
-	End                time.Time
+	Release            string              `json:"release" yaml:"release"`
+	PullRequestOptions *PullRequestOptions `json:"pull_request_options,omitempty" yaml:"pull_request_options,omitempty"`
+	Start              time.Time           `json:"start,omitempty" yaml:"start,omitempty"`
+	End                time.Time           `json:"end,omitempty" yaml:"end,omitempty"`
+}
+
+// ComponentReportRequestRelativeReleaseOptions is an unfortunate necessity for views where we do not have
+// a fixed time, rather a relative time to now/ga. It is translated to the above normal struct before use.
+type ComponentReportRequestRelativeReleaseOptions struct {
+	Release            string              `json:"release" yaml:"release"`
+	PullRequestOptions *PullRequestOptions `json:"pull_request_options,omitempty" yaml:"pull_request_options,omitempty"`
+	RelativeStart      string              `json:"start,omitempty" yaml:"start,omitempty"`
+	RelativeEnd        string              `json:"end,omitempty" yaml:"end,omitempty"`
```

Comment:
This made me sad but I tried it several different ways, it seemed the cleanest to layer in the relative time parsing for a view and yet always have a time.Time available when we get into component_report.go.